javascript/implementation/Database.ts

```diff
@@ -81,10 +73,30 @@ export class Database {
                 break;
             }
         }
-        if (!this.myChain) {
-            await this.startChain(identity);
+        if (reused) {
+            ensure(reused.medallion > 0);
+            this.myChain = reused;
+        } else {
+            const medallion = makeMedallion();
+            const chainStart = generateTimestamp();
+            const bundler = new Bundler(this.identity, medallion);
+            const bundleInfo = bundler.seal({
+                medallion, timestamp: chainStart, chainStart
+            });
+            ensure(bundleInfo.comment == this.identity);
+            const commitBytes = bundler.bytes;
+            await this.store.addBundle(commitBytes, true);
+            this.myChain = (await this.store.getClaimedChains()).get(medallion);
         }
-        ensure(this.myChain.medallion > 0);
+        this.iHave = await this.store.getChainTracker();
```

Comment:
It looks like you're setting `this.iHave` in `Database.initialize`.  You shouldn't need to grab it from the store again, only update the info you have in memory.